eclair-core/src/main/scala/fr/acinq/eclair/json/JsonSerializers.scala

```diff
@@ -472,22 +473,21 @@ object CustomTypeHints {
     classOf[MessageReceivedJson] -> "onion-message-received"
   ))
 
-  val channelStates: ShortTypeHints = ShortTypeHints(
-    List(
-      classOf[Nothing],
-      classOf[DATA_WAIT_FOR_OPEN_CHANNEL],
-      classOf[DATA_WAIT_FOR_ACCEPT_CHANNEL],
-      classOf[DATA_WAIT_FOR_FUNDING_INTERNAL],
-      classOf[DATA_WAIT_FOR_FUNDING_CREATED],
-      classOf[DATA_WAIT_FOR_FUNDING_SIGNED],
-      classOf[DATA_WAIT_FOR_FUNDING_LOCKED],
-      classOf[DATA_WAIT_FOR_FUNDING_CONFIRMED],
-      classOf[DATA_NORMAL],
-      classOf[DATA_SHUTDOWN],
-      classOf[DATA_NEGOTIATING],
-      classOf[DATA_CLOSING],
-      classOf[DATA_WAIT_FOR_REMOTE_PUBLISH_FUTURE_COMMITMENT]
-    ), typeHintFieldName = "type")
+  val channelStates: CustomTypeHints = CustomTypeHints(Map(
```

Comment:
I was forced to move to a `CustomTypeHints` because with `ShortTypeHints` we were now getting `ChannelStateData$DATA_*` strings which is really ugly...